### PR TITLE
Feature/444 - Adding entry point for querying user roles

### LINKED ISSUE #444: Add userRoles entry point
We should add an entry into the Graph for userRoles.

The resolver should limit returning data to authenticated users by default. 

A query like: 
```
{
  userRoles {
    id
    name
    capabilities
  }
}
```

Should return something like: 

```
{
  data: {
    userRoles: [
       {
         id: "cm9sZTphZG1pbg=="
         name: "admin",
         capabilities: [
			"edit_posts",
            "moderate_comments",
            ...
         ],
       }
    ]
  }
}
```

This should also support filtering by specific role: 

```
{
  userRoles( where: { name: "admin" } ) {
    ...RoleFields
  }
}
```

### PR DESCRIPTION
**Pull request by CodeProKid**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [x] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
Implements queries for User Roles

Does this close any currently open issues?
------------------------------------------
closes #444 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
Example query for getting all roles:
```graphql
query {
	userRoles{
		edges {
			node {
				name
				capabilities
				id
			}
		}
	}
}
```
Example query for getting info on a single role:
```graphql
query {
	userRole(id: "cm9sZTphZG1pbmlzdHJhdG9y"){
		id
		name
		capabilities
	}
}
```

Any other comments?
-------------------
There are some new autoloader files in here that weren't previously in the repo. I don't think I did anything weird to have these generated, not sure if it's just because I'm on a different version of composer or what though.

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** 4.9.6

### CODE REVIEW

**Comment by CodeProKid** [comment]
Sooooo, I ran `composer dump-autoload` and I think that somehow made things worse? 😆 looks like it's loading the dev deps too 😬

**Comment by jasonbahl** [comment]
@CodeProKid try running: 

```
rm -rf vendor
rm -rf composer.lock
composer install --no-dev
```

**Comment by CodeProKid** [comment]
@jasonbahl looks like build is passing now. Wanna take a look when you get a chance?

File: composer.lock [hunk 1]
```diff
@@ -7376,4 +7376,4 @@
     "prefer-lowest": false,
     "platform": [],
     "platform-dev": []
-}
```
Comment (jasonbahl) [inline composer.lock]: 😭

File: src/Data/DataSource.php [hunk 2]
```diff
@@ -371,6 +372,34 @@ public static function resolve_users_connection( $source, array $args, $context,
...
+			throw new UserError( sprintf( __( 'No user role was found with the name %s', 'wp-graphql' ), $name ) );
+		} else {
+			$role = (array) $role;
+			$role['id'] = $name;
+			return $role;
+		}
+
+	}
+
+	public static function resolve_user_role_connection( $source, array $args, AppContext $context, ResolveInfo $info ) {
```
Comment (hughdevore) [inline src/Data/DataSource.php]: Need a doc block here

**Comment by jasonbahl** [comment]
I'm pulling this locally to check some things out as well.

File: composer.lock (hunk 1 above)
Comment (CodeProKid) [reply composer.lock]: Yeah, my IDE has been doing that on some files. I'll check it out.

**Comment by CodeProKid** [comment]
@hughdevore & @jasonbahl this should be fixed up now 👍

**Comment by jasonbahl** [comment]
@CodeProKid good work. got this merged 😄

**Pull request merged**